core/src/avm2/globals/flash/text/textformat.rs

```diff
@@ -641,3 +634,23 @@ pub fn create_class<'gc>(mc: MutationContext<'gc, '_>) -> GcCell<'gc, Class<'gc>
 
     class
 }
+
+/// Implements the IEEE-754 "Round to nearest, ties to even" rounding rule.
+/// (e.g., both 1.5 and 2.5 will round to 2).
+/// This is the rounding method used by Flash for the above coercions.
+/// Although this is easy to do on most architectures, Rust provides no standard
+/// way to round in this manner (`f64::round` always rounds away from zero).
+/// For more info and the below code snippet, see: https://github.com/rust-lang/rust/issues/55107
+/// This also clamps out-of-range values and NaN to `i32::MIN`.
+/// TODO: Investigate using SSE/wasm intrinsics for this.
+fn round_to_i32(x: f64) -> i32 {
```

Comment:
You left this duplicated :)

Reply:
Yeah, certainly we need to put this function in some shared place, but I'm not sure where. Also note that `core/src/matrix.rs` has a similar method as well (which I copied and changed a bit).